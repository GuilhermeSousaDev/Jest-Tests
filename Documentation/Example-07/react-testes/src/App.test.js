import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const component = renderer.create(
    <App page="http://www.facebook.com">Facebook</App>
  );
  
  let tree = component.toJSON();

  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
