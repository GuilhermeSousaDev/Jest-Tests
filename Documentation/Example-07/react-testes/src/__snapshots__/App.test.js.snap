// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders learn react link 2`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
